identity/src/main/java/com/android/identity/trustmanagement/TrustManager.kt

```diff
@@ -0,0 +1,172 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.trustmanagement
+
+import com.android.identity.storage.StorageEngine
+import org.bouncycastle.asn1.x500.X500Name
+import java.io.ByteArrayInputStream
+import java.security.cert.CertificateFactory
+import java.security.cert.PKIXCertPathChecker
+import java.security.cert.X509Certificate
+
+/**
+ * This class is used for the verification of a certificate chain
+ */
+class TrustManager(val certificateStorage: StorageEngine) {
+
+    private val certificates: MutableMap<X500Name, X509Certificate>
+
+    /**
+     * Nested class containing the result of the verification of a certificate chain
+     */
+    class TrustResult(
+        var isTrusted: Boolean,
+        var trustChain: List<X509Certificate> = emptyList(),
+        var error: String = ""
+    )
+
+    /**
+     * Initialize private variables
+     */
+    init {
+        certificates = getAllCertificates().associateBy { X500Name(it.subjectX500Principal.name) }
+            .toMutableMap()
+    }
+
+    /**
+     * Save a certificate
+     */
+    fun saveCertificate(certificateBytes: ByteArray) {
+        val certificate = parseCertificate(certificateBytes)
+        if (certificateExists(certificate)) {
+            throw Exception("File already exists")
+        }
+        val name = X500Name(certificate.subjectX500Principal.name)
+        certificateStorage.put(name.toString(), certificate.encoded)
+        certificates[name] = certificate
+    }
+
+    /**
+     * Check that a certificate exists
+     */
+    fun certificateExists(certificate: X509Certificate): Boolean{
+        val name = X500Name(certificate.subjectX500Principal.name)
+        return certificateStorage.get(name.toString()) != null
+    }
+
+    /**
+     * Get all the certificates in the [TrustManager]
+     */
+    fun getAllCertificates(): List<X509Certificate> {
+        return certificateStorage.enumerate().map {
+            parseCertificate(certificateStorage.get(it)!!)
+        }
+    }
+
+    /**
+     * Delete a certificate
+     */
+    fun deleteCertificate(certificate: X509Certificate){
+        val name = X500Name(certificate.subjectX500Principal.name)
+        certificateStorage.delete(name.toString())
+        certificates.remove(name)
+    }
+
+    /**
+     * * Verify a certificate chain (without the self-signed root certificate) by mDoc type with
+     * the possibility of custom validations on the certificates (mdocAndCRLPathCheckers),
+     * for instance the mDL country code
```

Comment:
This sentence has some issues

- Nit: Superflous `*` in first line
- It says `by mDoc type' but is there anything mdoc specific here? 
- No idea what "(mdocAndCRLPathCheckers)" refers to
- "for instance the mDL country code" .. I believe this refers to `CountryValidator`. If so, link to it.
- Missing whitespace before `@param` below.

Reply:
The code was refactored, the comment still not...